File: benchmarks/baselines/c/state_machine.c
/* FLOW benchmark baseline. Human decisions: 5 (states, events, initial, transitions, terminal). */
#include <stdio.h>
#ifdef FLOW_BENCHMARK
static volatile const char *flow_benchmark_last_format;
static void flow_benchmark_sink(const char *format, ...) { flow_benchmark_last_format = format; }
#define flow_printf(...) flow_benchmark_sink(__VA_ARGS__)
#else
#define flow_printf printf
#endif
int main(void) {
    enum State { IDLE, RUNNING, DONE }; enum State state = IDLE; int events[] = {1,1,2,2};
    for (int i = 0; i < 4; ++i) { if (state == IDLE && events[i] == 1) state = RUNNING; else if (state == RUNNING && events[i] == 2) state = DONE; }
    flow_printf("final_state %d\n", state); return state == DONE ? 0 : 1;
}

File: benchmarks/baselines/rust/state_machine.rs
// FLOW benchmark baseline. Human decisions: 5.
#[derive(Clone, Copy)] enum State { Idle, Running, Done }
use std::env;
fn main() { let reps = env::var("FLOW_BENCH_REPS").ok().and_then(|v| v.parse().ok()).unwrap_or(1); for _ in 0..reps { let mut s = State::Idle; for e in [1,1,2,2] { s = match (s,e) { (State::Idle,1) => State::Running, (State::Running,2) => State::Done, _ => s }; } std::hint::black_box(s); } }
